### DISCUSSION TITLE
Welcome to constellation-ui-gallery Discussions!

### DISCUSSION THREAD

**Discussion by ricmars** [pegasystems/constellation-ui-gallery#22] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by TWM-PDL** [comment]
I’m working on a project where we intend to use components from the Constellation UI Gallery. Could anyone provide information on who is responsible for maintaining these components? Specifically, once I use the components, does the responsibility for maintenance shift to the end user, or will the components be upgraded automatically as new releases become available?

Apologies if this is a basic question; it’s my first time using this library.

**Reply by ricmars** [reply]
this is an open source project based on an Apache 2 license https://github.com/pegasystems/constellation-ui-gallery?tab=Apache-2.0-1-ov-file#readme. Here is an extract of the license: "software  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.".

In term of support and maintenance, it follow the same approach as any other open source libraries - code is delivered as IS and since you are importing the components in your application ruleset, you are responsible for this code inside your application. The source code is provided and you can make changes for your application and fork the code. We are providing release version for 23.1 and 24.1 but only the latest Pega Platform version will be actively worked on. 

You can submit bugs, ideas using the following link: https://github.com/pegasystems/constellation-ui-gallery/issues
You can contribute to this repo and add new components or extend existing ones - see https://github.com/pegasystems/constellation-ui-gallery/blob/master/CONTRIBUTING.md

**Comment by rajsb3038** [comment]
Hi Richard, do we have constellation UI Gallery for Infinity 24.2? In the computer land application compare table view is not displaying. I am using ConstellationUIGallery 2.0.1 which is compatible for 24.1 but my current cloud environment is 24.2.

**Comment by MandeepBIG** [comment]
Hi @ricmars,

I am using the constellation-ui-gallery master repository and trying to run it on my local machine. However, when I execute the npm start command, I encounter an error in the browser: and also I added a screen shot for Ref.

"Oh no! Your Storybook is empty. Possible reasons why:"
The glob specified in main.js is incorrect.
No stories are defined in your story files.
You're using filter functions, and all stories are being filtered out.
![DxComponentStoryBook](https://github.com/user-attachments/assets/9008f48a-2b57-468d-b1d1-f4adefda8815)

In VS code it's showing error 
WARN No story files found for the specified pattern: src\**\*.mdx
WARN No story files found for the specified pattern: src\**\*.stories.@(js|jsx|mjs|ts|tsx)

![DxComponent](https://github.com/user-attachments/assets/052dacf4-6f77-4978-a486-d5c987fd85dc)

for Ref. I added a screenshot for folder structure and main.js file.

**i am using Node version:- v20.18.1 and NPM version:- 10.8.2**

Could you please help me troubleshoot this issue?

**Comment by Revanth030303** [comment]
hi @ricmars 

We are using Pega 24.1.3 version in our project and we are using Case Launcher widget in one of our landing page
https://github.com/pegasystems/constellation-ui-gallery/tree/master/src/components/Pega_Extensions_CaseLauncher

Our client is expecting the position of the Button to be on left hand side instead of right hand side

Can you please help on like adding a param for repositioning, so that we can choose left or right from our end in the widget.

Please check the attachment for better understanding

<img width="923" height="90" alt="image" src="https://github.com/user-attachments/assets/6a5ebd40-4b3f-4102-9676-9bc53a885580" />

Thanks,
Revanth

**Reply by ricmars** [reply]
the fix is a one line code change + some config json <CardFooter justify='end'> - You can submit a branch against 2.x for this enhancement or donwload the 2.x code and make the code change directly and publish to your application.

**Comment by satkc** [comment]
Hi,

In our application (v24.2.1, on-prem server) using DX components, the Status Badge occasionally displays blank values. After saving the Application rule, the field data appears correctly. Looking for a fix to ensure the Status Badge consistently displays data without requiring a save.

**Reply by satkc** [reply]
Could someone please assist with this issue?

**Comment by bararA9** [comment]
Hi @ricmars ,

We are looking for a UX improvement. The Case Launcher Widget from the Constellation UI Gallery does not include built-in busy indicator functionality during case creation operations. 

For our WSS portals, we are using Case Launcher Widget. 

Is it possible to extend the "Case Launcher Widget" to contain the built-in busy indicator (or deferload)? It could help everyone:
1. If there are some actions taking a longer time, the busy indicator makes sure that the create button is not clicked again.
2. This will make the behavior consistent across all case creation options within the WSS and Standard portals.

Please do let me know if that sounds like a good idea ?

Kind Regards,
Ashish
